### ISSUE TITLE
`generate` and `train_step` hardcode audio offset 260 instead of deriving from config

### ISSUE THREAD

**Issue by ngo-software** [fufuchiu/speechloom#24]
In `src/speechloom/model.py`, both `generate` and `train_step` embed the magic
number `260` directly:

- `generate`: `logits[260:] = -torch.inf` (text mode) and `logits[3:260] = -torch.inf` (audio mode)
- `train_step`: `joint_loss(model(audio, lengths, input_ids), labels, 260, ...)`

The value 260 is `4 reserved tokens + 256 UTF-8 byte values`, which is correct
today. But `SpeechConfig` does not expose this as a property, so every call
site re-derives it by convention.

`TokenLayout` already has `audio_offset` as a property. If `SpeechConfig`
gained the same property, these call sites could use
`model.config.audio_offset` and stay correct if the reserved-token layout ever
changes.

This is not a bug today -- the numbers are right -- but it is a maintenance
hazard. Anyone adding a new reserved token or changing the text byte range
would need to hunt down every `260` literal in `model.py`.

Proposed fix:

```python
class SpeechConfig:
    @property
    def audio_offset(self) -> int:
        return 260
```

Then replace the three `260` literals in `generate` and `train_step` with
`self.config.audio_offset` / `model.config.audio_offset`.
